### ISSUE TITLE
Should stateless component be ignored for the multi-component lint rule?

### ISSUE THREAD

**Issue by matthargett** [callstack/eslint-config-callstack#50]
## Ask your Question

I am running into a scenario where I declare an HOC component, and then export it using React.forwardRef() in the same file. This is triggering an error that I have 2 components declared in 1 file.

This option can be disabled for essentially HOC components:
```json
"react/no-multi-comp": [<enabled>, { "ignoreStateless": <boolean> }]
```
I'd like to propose adding that tweak. Otherwise, can it be explained how to resolve the rule firing when using forwardRef()?

**Comment by thymikee** [comment]
I'm not a fan of this rule honestly and would like to just disable it (wasn't aware it was turned on really).

In my experience the teams I'm working with, having this rule enabled, use to create more `_renderSomething` class methods to achieve something that would be extracted to a small local component (and maybe to a reusable one sometime later easily), resulting in more imperative code.

What do you think?

cc @satya164 @michalchudziak @ferrannp

**Comment by satya164** [comment]
Yes lets disable it

**Comment by ferrannp** [comment]
I'm ok with disabling it.

**Comment by matthargett** [comment]
Submitted PR #51
